mapslang/Makefile

```diff
@@ -1,27 +1,83 @@
-# Project: adikted
-
-#To create a debug executable, run with the command:
-# make DEBUG
-
-#ON_LINUX = 1
-
-ifneq (,$(ON_LINUX))
+# Project: mapslang
+# Makefile created by KeeperFX/dkfans team
+
+# Instructions:
+# To create the Windows executable run:
+# make all
+# To create the linux executable run:
+# make all BUILD_FOR_LINUX=1 (not fully implemented)
+# To create a debug executable, run with the command:
+# make all DEBUG=1
+# To clean up the built files (objects, library, and resource file) run:
+# make clean
+
+# Windows version can be built with:
+# - Windows   : MSYS2 (MinGW 32-bit) - make
+# - Windows   : MSYS2 (MinGW 32-bit) - mingw32-make
+# - Windows   : MinGW (MSYS) - make
+# - WSL/Linux : i686-w64-mingw32-gcc - make
+
+# Executable file extensions on target environment
+BUILD_FOR_LINUX ?= 0
+ifeq ($(BUILD_FOR_LINUX), 1)
+  EXEEXT =
+  OTHEREXEEXT = .exe
+  LIBEXT = .so
+  OTHERLIBEXT = .dll
+  SLANGPLATFORM =
+else
+  EXEEXT = .exe
+  OTHEREXEEXT =
+  LIBEXT = .dll
+  OTHERLIBEXT = .so
+  SLANGPLATFORM = WIN32 MINGW32 DLL
+endif
+# Host environment
+ifneq (,$(findstring Windows,$(OS)))
+  CROSS_EXEEXT = .exe
+  ifneq ($(BUILD_FOR_LINUX), 1)
+   SLANGBUILDCOMMAND = mkfiles/build.sh $(SLANGPLATFORM)
+  else
+   SLANGBUILDCOMMAND =
+  endif
+else
+  CROSS_EXEEXT =
   CROSS_COMPILE = i686-w64-mingw32-
-  SLANG_HOST=mingw32
+  ifneq ($(BUILD_FOR_LINUX), 1)
+    SLANGBUILDCOMMAND = cp ./src/slconfig.h ./src/config.h && \
+                        gcc -I./src ./src/mkfiles/mkmake.c ./src/slprepr.c -o ./src/mkfiles/mkmake && \
+                        src/mkfiles/mkmake WIN32 MINGW32 DLL     < src/mkfiles/makefile.all >     src/Makefile && sed -i 's/TOOL_PREFIX =/TOOL_PREFIX = $(CROSS_COMPILE)/g'      src/Makefile && \
```

Comment:
Any suggestions for cleaning this section up? It is essentially `/lib/slang-pre2.3.3-24/mkfiles/build.sh`  but `TOOL_PREFIX = $(CROSS_COMPILE)` needs to be added to the 3 Makefiles.

Should I create a new shell script? Or is there a clever way to do it in one line with the original .sh script?

Reply:
You may use `sed` or `patch` on them.

Reply:
patch files created in `/mapslang/libslang-patches` and used by `mapslang/Makefile`